### DISCUSSION TITLE
Run different modes from within Rofi

### DISCUSSION THREAD

**Discussion by LimCharles** [davatorium/rofi#1900] [Ideas/Feature Requests]
I wanted to make a simple script to run Rofi where the different modes would be listed, such as run, filebrowser, ssh, window, etc., but I always ran into the error: "Do not run rofi from within rofi."

I was wondering if this was a feature anyone else was searching for, or if anyone found a solution by calling the sourcecode (run.c, etc.).

**Comment by DaveDavenport** [comment]
what do you mean with 'calling the source code'?

```bash
rofi -show drun -modes 'run,drun,ssh,window,keys'
```

you can then switch between modes in rofi, even get a list of the modes.
Like here on top:
![rofi-2023-09-26-2004-00001](https://github.com/davatorium/rofi/assets/406245/8a14eb98-95c2-4031-85ea-ee4a2040d94f)

**Reply by LimCharles** [reply]
Oh that's much smarter than what I wanted! I made a script that echoed each mode (run, window, ssh) then I tried to run another instance of Rofi when it was chosen. Thanks for that!
